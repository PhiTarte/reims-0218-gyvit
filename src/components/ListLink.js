// TODO list of links to navbar

import React from "react"
import Link from "./Link"
import { Nav, NavItem, Navbar, Row } from "reactstrap"

const ListLink = ({ dataLinks }) => (
  <Navbar
    light
    expand="md"
    style={{
      width: "220px",
      height: "1024px",
      "background-color": "#231b56"
    }}
  >
    <Nav vertical>
      <NavItem>
        {dataLinks.map((link, index) => <Link key={index} {...link} />)}
      </NavItem>
    </Nav>
  </Navbar>
)

export default ListLink
