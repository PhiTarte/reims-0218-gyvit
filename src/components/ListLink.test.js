import React from "react"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"
import ListLink from "./ListLink"

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<ListLink />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it("renders link html", () => {
  const tree = renderer.create(<ListLink />).toJSON()
  expect(tree).toMatchSnapshot()
})
