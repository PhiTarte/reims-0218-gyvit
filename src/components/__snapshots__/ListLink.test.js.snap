// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders link html 1`] = `
<div
  className="row"
>
  <div
    className="col-3"
  >
    <nav
      className="navbar navbar-expand-md navbar-light bg-dark"
    >
      <ul
        className="nav flex-column"
      >
        <li
          className="nav-item"
        >
          <div>
            <a
              className="nav-link"
              href="#"
              onClick={[Function]}
            >
              <img
                alt=""
                src="/img/house_38533.ico"
              />
              Home
            </a>
            <hr />
          </div>
          <div>
            <a
              className="nav-link"
              href="#"
              onClick={[Function]}
            >
              <img
                alt=""
                src="/img/files.ico"
              />
              Files
            </a>
            <hr />
          </div>
          <div>
            <a
              className="nav-link"
              href="#"
              onClick={[Function]}
            >
              <img
                alt=""
                src="/img/share.ico"
              />
              Shares
            </a>
            <hr />
          </div>
          <div>
            <a
              className="nav-link"
              href="#"
              onClick={[Function]}
            >
              <img
                alt=""
                src="/img/templates.ico"
              />
              Templates
            </a>
            <hr />
          </div>
          <div>
            <a
              className="nav-link"
              href="#"
              onClick={[Function]}
            >
              <img
                alt=""
                src="/img/request.ico"
              />
              Requests
            </a>
            <hr />
          </div>
          <div>
            <a
              className="nav-link"
              href="#"
              onClick={[Function]}
            >
              <img
                alt=""
                src="/img/links.ico"
              />
              Public links
            </a>
            <hr />
          </div>
          <div>
            <a
              className="nav-link"
              href="#"
              onClick={[Function]}
            >
              <img
                alt=""
                src="/img/contact.ico"
              />
              Contacts
            </a>
            <hr />
          </div>
          <div>
            <a
              className="nav-link"
              href="#"
              onClick={[Function]}
            >
              <img
                alt=""
                src="/img/settings.ico"
              />
              Settings
            </a>
            <hr />
          </div>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
